### PR TITLE
Valet Reservation List Refactor + Routing

### PR DESCRIPTION
**Pull request by kjmak14**
Added routing to ReservationDetails page and refactored reservation list to include redux

### CODE REVIEW

**Review by kssampson — APPROVED** [review]
Thanks for starting in to refactor in redux. We shouldn't have merge conflicts after changing the window.location.href in Scanner. Coming along!

File: src/components/valet-reservation/ReservationList.jsx [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 import getReservationList from "../../utils/getReservationList";
+
 const ReservationList = () => {
-  const [reservations, setReservations] = useState([]);
-  const [currentHour, setCurrentHour] = useState(new Date().getHours());
   const[showNextHour, setShowNextHour] = useState(false);
+  const { time, reservations } = useSelector((state) => state.valet);
+  const dispatch = useDispatch();
 
   const fetchReservations = async () => {
```
Comment (bzw24) [inline src/components/valet-reservation/ReservationList.jsx:12]: do you plan to make this more dynamic? Some information here are hard coded

File: src/components/valet-reservation/ReservationList.jsx [hunk 2]
```diff
@@ -26,20 +30,18 @@ const ReservationList = () => {
 
 useEffect(() => {
```
Comment (bzw24) [inline src/components/valet-reservation/ReservationList.jsx:31]: Why is there 3 useEffects?? Is there anyway to consolidate it into one and refactor them? Seems like this useEffect is relying on the bottom useEffect for setting time, then you are checking for changes in time. Can you verify whether this  goes into infinite loop?

**Review by bzw24 — COMMENTED** [review]
Left some comments, would advise on some refactoring the logic on your reservation List page

**Review by stevkim — APPROVED** [review]
Base code looks good! 
I talked to kyle s. about potentially refactoring a few thing tomorrow! 

But great job getting redux working!

**Pull request merged**